projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -462,6 +462,7 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
             selectAllBtn.indeterminate = anyFiltered && anyUnfiltered;
             if (this.isHierarchical() && this.tree) {
                 this._hierarchicalSelectedItems = this.tree.nodes.map(n => n.data as FilterListItem).filter(item => item.isFiltered);
+                this.tree.collapseAll();
```

Comment:
Okay, not the usual fix. Bit odd too, not sure how collapsing the nodes helps and wouldn't this change actual behavior?
See https://www.infragistics.com/products/ignite-ui-angular/angular/components/treegrid/excel-style-filtering and filter the second column, if anything the tree auto-expands for matches
![image](https://github.com/user-attachments/assets/7d16ca82-0dbf-44ad-8aec-5b74dc580004)

Reply:
@damyanpetev It actually retains the old behavior, which is when you clear the search all those nodes will get collapsed. In this branch, probably because the nodes don't get destroyed/re-created anymore, they stay expanded - hence breaking this test: 
https://github.com/IgniteUI/igniteui-angular/actions/runs/13324852547/job/37215848062?pr=14864#step:11:11465

Reply:
Oh, okay, had to expand more on top to see it's on empty search. Yeah, makes more sense in this case.
The `.filter(item => item.isFiltered)` completely threw me off 😆